src/emulator/util/include/util/log.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#define SPDLOG_WCHAR_FILENAMES
+#define SPDLOG_NO_THREAD_ID
+#include "spdlog/spdlog.h"
+
+#ifdef _MSC_VER
+#include "spdlog/sinks/msvc_sink.h"
+#endif
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+// needed for cross-platform getcwd
+#ifdef _WIN32
+#include <direct.h>
+#define getcwd _getcwd // stupid MSFT "deprecation" warning
+#else
+#include <unistd.h>
+#endif
```

Comment:
The directories to put things in are available from SDL. We want the "pref path."

Currently `host`'s `init` function puts it in HostState, but we could get it earlier if we need to.

Reply:
Wouldn't this put stuff in AppData on Windows? As a Windows user, that's terrible
It's convenient at all, in fact I'd expect the uma0 and ux0 folders to be in the same directory as the emulator too, or at least close to it, but that's another discussion.

These are all files that someone using this emulator needs to have access to, and this directory is hard to reach.